### PR TITLE
Progress Redesign with PLC/Peer Reviews

### PR DESCRIPTION
**Pull request by Bjvanminnen**
This PR makes it so that we support focus areas/peer reviews for PLC in the progress redesign.

Focus area in detail view
image

Focus area in summary view:
image

Peer reviews in detail view:
image

Peer reviews in summary view:
image

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
Brad, if you could look at React/redux stuff. Mehal, if you want to weigh in on any of the PLC aspects of this (though I'm not sure you're as familiar with the client side stuff).

File: apps/src/code-studio/progressRedux.js [hunk 1]
```diff
@@ -322,6 +355,17 @@ export const categorizedLessons = state => {
     byCategory[category].levels.push(stageLevels);
   });
 
+  // Peer reviews get their own category, but these levels/lessson are stored
+  // separately from our other levels/lessons in redux (since they're slightly
+  // different)
+  if (state.peerReviewStage) {
+    byCategory['Peer Review'] = {
+      category: 'Peer Review',
```
Comment (islemaster) [inline apps/src/code-studio/progressRedux.js:363]: Do we need i18n on PLC/peer review views?

File: apps/src/templates/progress/FocusAreaIndicator.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
+/**
+ * A component that adds a sort of ribbon looking focus indicator to the side
+ * of your row. Note: This is English only.
```
Comment (islemaster) [inline apps/src/templates/progress/FocusAreaIndicator.jsx]: Ah, answers my question from before.

Also, no need to sound hesitant here. "A focus indicator ribbon for the side of your row."

File: apps/src/code-studio/progressRedux.js (hunk 1 above)
Comment (Bjvanminnen) [reply apps/src/code-studio/progressRedux.js:363]: My understanding is no (and I'm pretty sure we don't get it currently).

File: apps/src/templates/progress/FocusAreaIndicator.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    display: 'inline-block',
+    position: 'relative',
+    top: 10
+  },
+  leftArrow: {
+    display: 'inline-block',
+    width: 0,
+    height: 0,
+    borderStyle: 'solid',
+    borderWidth: '17px 20px 17px',
```
Comment (islemaster) [inline apps/src/templates/progress/FocusAreaIndicator.jsx:30]: Doesn't this shorthand to `'17px 20px'`?

File: apps/src/templates/progress/FocusAreaIndicator.jsx (hunk 2 above)
Comment (Bjvanminnen) [reply apps/src/templates/progress/FocusAreaIndicator.jsx]: Heh. I think the hesitancy came from the fact that it looks somewhat less ribbony than the spec ;-)

File: apps/src/templates/progress/FocusAreaIndicator.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    display: 'inline-block'
+  },
+  focusAreaIcon: {
+    display: 'inline-block',
+    color: color.white,
+    lineHeight: '13px',
+    padding: 5,
+    marginLeft: 5,
+    marginRight: 5,
+    borderRadius: 4,
```
Comment (islemaster) [inline apps/src/templates/progress/FocusAreaIndicator.jsx:52]: Hooray for automatic vendor prefixing via Radium :+1:

File: apps/src/templates/progress/FocusAreaIndicator.jsx (hunk 3 above)
Comment (Bjvanminnen) [reply apps/src/templates/progress/FocusAreaIndicator.jsx:30]: Nope. No left border (that's how we get a triangle).

File: apps/src/templates/progress/ProgressBubble.jsx [hunk 5]
```diff
@@ -53,7 +53,9 @@ const ProgressBubble = React.createClass({
   },
 
   render() {
-    const { number, status, url, disabled, levelName, levelIcon } = this.props;
+    const { number, status, url, levelName, levelIcon } = this.props;
+
+    let disabled = this.props.disabled || levelIcon === 'lock';
```
Comment (islemaster) [inline apps/src/templates/progress/ProgressBubble.jsx]: Nit: Can be `const`

File: apps/src/templates/progress/ProgressTable.jsx [hunk 6]
```diff
@@ -59,27 +58,6 @@ const ProgressTable = React.createClass({
...
-                }
-              ]}
-              levelsByLesson={[[{
-                status: 'not_tried',
-                url: '',
-                name: 'Not Implemented'
-              }]]}
-            />
-          }
         </div>
```
Comment (islemaster) [inline apps/src/templates/progress/ProgressTable.jsx:61]: Nice to see some special cases go away (or at least get pushed down)

File: apps/src/code-studio/progressRedux.js (hunk 1 above)
Comment (mehalshah) [reply apps/src/code-studio/progressRedux.js:363]: correct

**Pull request merged**